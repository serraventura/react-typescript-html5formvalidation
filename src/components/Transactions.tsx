import React, { useContext } from "react";
import { Text, Header3 } from "@zopauk/react-components";
import { Container, Scrollable, CardFlex } from "./styled";
import { ThemeContext } from "./ThemeContext";
import { ITransactions } from "./types";

const Transactions: React.FC<ITransactions> = ({ list = [] }) => {
  const context = useContext(ThemeContext);
  return (
    <Container color={context.neutral.neutral50}>
      <Header3>Transactions</Header3>
      <Scrollable>
        {!!list.length ? (
          list.map(i => (
            <CardFlex key={`${i.email}${i.name}`}>
              <div>
                <Text fw="bold" size="xl">
                  {i.name}
                </Text>
                <br />
                <Text size="s">{i.email}</Text>
              </div>
              <div>
                <Text fw="bold" size="xl">
                  Â£{i.amount}
                </Text>
              </div>
            </CardFlex>
          ))
        ) : (
          <CardFlex
            borderColor={context.neutral.neutral50}
            backgroundColor={context.neutral.neutral75}
          >
            No transactions
          </CardFlex>
        )}
      </Scrollable>
    </Container>
  );
};

export default Transactions;
