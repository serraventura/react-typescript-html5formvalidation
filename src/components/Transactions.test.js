import React from "react";
import { shallow, mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Transactions from "./Transactions";

configure({ adapter: new Adapter() });

const transactions = {
  name: "john",
  email: "john@mail",
  amount: 0
};

describe("<Transactions />", () => {
  it("should render without crashing", () => {
    const wrapper = shallow(<Transactions />);
    expect(wrapper).toBeTruthy();
  });

  it("should match snapshot", () => {
    const wrapper = mount(<Transactions />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render with data without crashing", () => {
    const wrapper = mount(<Transactions list={transactions} />);
    expect(wrapper).toBeTruthy();
  });

  it("should match snapshot with data", () => {
    const wrapper = mount(<Transactions list={transactions} />);
    expect(wrapper).toMatchSnapshot();
  });
});
