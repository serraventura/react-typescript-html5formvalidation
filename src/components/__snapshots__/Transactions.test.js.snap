// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Transactions /> should match snapshot 1`] = `ReactWrapper {}`;

exports[`<Transactions /> should match snapshot with data 1`] = `ReactWrapper {}`;
